### PR TITLE
refactor: move node-core/engine to standalone crate

### LINKED ISSUE #9099: move node-core/engine to standalone crate
### Describe the feature

this is responsible for a lot of bloat

https://github.com/paradigmxyz/reth/blob/270df3eeb807e148d317d0037b2bf1499530a1d1/crates/node-core/src/engine/mod.rs#L1-L1

and should be moved to `crates/engine/util`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #9099

### CODE REVIEW

File: crates/engine/utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "reth-engine-utils"
```
Comment (mattsse) [inline crates/engine/utils/Cargo.toml:2]:
```suggestion
name = "reth-engine-util"
```

File: crates/node-core/Cargo.toml [hunk 2]
```diff
@@ -69,7 +66,7 @@ once_cell.workspace = true
 # io
 dirs-next = "2.0.0"
 shellexpand = "3.0.0"
-serde.workspace = true
+# serde.workspace = true
```
Comment (mattsse) [inline crates/node-core/Cargo.toml:69]:
```suggestion
```

**Review by mattsse — CHANGES_REQUESTED** [review]
nice,

very pedantic naming nit :) utils -> util

**Comment by tcoratger** [comment]
> nice,
> 
> very pedantic naming nit :) utils -> util

@mattsse ahah I saw that you put `util` in the issue but I thought it was a typo like I'm more used to seeing `utils`, no problem, just fixed.

By the way, I also saw that there was an engine-primitives crate, shouldn't we move it to engine/primitives?

**Review by mattsse — APPROVED** [review]

**Pull request merged**